docs/css-prop.mdx

```diff
@@ -245,6 +245,9 @@ The styles are concatenated together and inserted via `insertRule`.
 
 Relying on the css spec's ["Order of Appearance"](https://www.w3.org/TR/css-cascade-3/#cascade-order) rule, property values defined later (green) override those before it (red).
 
+If you find that emotion's default style precedence does not work for your use-case, you can override it with the [Class Names API]/docs/class-names). 
```

Comment:
oh you're right! I'm in a situation where i have a render prop defining some styles which i want to be able to amend as a consumer. 

For example: 
```jsx
<ClassNames>
  {({ css }) => {
    const props = {
      className: css`... my base styles here`
     };
     
     return children(props)
 }}
</ClassNames>
```

And the consumer would do something like:

```jsx
<MyComponent>
  {({ className }) => {
    return <span className={className} css={css`..my custom styles here`}>Foo bar</span>
  }}
</MyComponent>
```

In this case, the custom styles are applied before and overridden by the base styles. I can't really find another way around it 🤔

Reply:
I'm not entirely sure but I think this might work:
```js
<MyComponent>
  {({ className }) => {
    return <span css={[className, css`..my custom styles here`]}>Foo bar</span>
  }}
</MyComponent>
```
It would be great if you could share a repro case, this would allow me to jump into this way quicker and testing things right away.

Reply:
Hey sure thing, this is what i'm thinking 😄 

https://codesandbox.io/s/naughty-haibt-g9khc

Reply:
Ok, thanks for the repro! I have a better understanding of the issue now.

So yea - basically your way is correct, although I'm wondering - why have you chosen this kind of a render prop API? 🤔 

You can also check out this discussion around css prop specificity, it might get you a better understanding of why things work like this:
https://github.com/emotion-js/emotion/issues/753

Reply:
Amazing thanks for the help!
I'll close this PR for now since i see now that it doesn't really make sense unless you're trying to solve my specific use-case 😄